nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongoRecord.java

```diff
@@ -131,7 +131,7 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 for (String name : schema.getFieldNames()) {
                     document.put(name, contentMap.get(name));
                 }
-                inserts.add(document);
+                inserts.add(convertArrays(document));
```

Comment:
For future-proofing, perhaps this method should be called something like normalizeFields() or something?

Reply:
I prefer convertArrays because it ties back to the ticket and AFAIK this is the only area where the Record API throws Mongo's client for a loop on the fields.